complex test case to find outer planar graph face

adamhosticka opened the pull request:
Finding outer face in a more complex graphs doesn't work.

adamant-pwn commented:
Hi, thanks for the pull request! But I'm a bit confused, so you mean that the current implementation is incorrect? Is it possible for you to also add a fix for it in such a case?

adamhosticka commented:
Yes, I believe there is an error. Unfortunately I do not have the time right now create a fix in cpp. I am coding in Python and managed to create a workaround, but I am not even sure it is correct. All I can provide right now is the comment I have added to my code 2 weeks ago:

> The sourced algorithm contains an error when finding the outer face. It cannot be just found by checking the orientation
    of the vertices. Therefore, I changed the logic.
    First we find the outer face candidates - these are the faces containing all the vertices.
    The outer face is then the face with the least vertices.
    Proof by dispute:
        The face with more vertices `f1` must contain at least one extra vertex `v`.
        This vertex lies inside the face with fewer vertices `f2`.
        If we create a new face `f3` by splitting any edge (a,b) of the `f2` into (a,v)&(v,b),
            then the edge (a,b) won't lie in the new face `f3`.
        Therefore, the extra vertex `v` is not a part of the outer face.
        Repeat for all extra vertices.
